File: analyze_excel.py
import openpyxl
import sys

try:
    wb = openpyxl.load_workbook('modelo_indicadores/g1_prazo_decorrido.xlsx')
    print(f"✅ Arquivo carregado com sucesso!")
    print(f"\n📋 Abas encontradas: {wb.sheetnames}")
    print(f"   Total de abas: {len(wb.sheetnames)}")
    
    for sheet_name in wb.sheetnames:
        sheet = wb[sheet_name]
        print(f"\n{'='*60}")
        print(f"📄 ABA: {sheet_name}")
        print(f"{'='*60}")
        
        # Ler primeiras 15 linhas
        for i, row in enumerate(sheet.iter_rows(max_row=15, values_only=True), 1):
            # Filtrar células vazias no final
            row_data = [cell for cell in row if cell is not None]
            if row_data:
                print(f"Linha {i}: {row_data}")
        
        print(f"\nTotal de linhas com dados: {sheet.max_row}")
        print(f"Total de colunas: {sheet.max_column}")
    
    wb.close()
    
except Exception as e:
    print(f"❌ Erro ao analisar arquivo: {e}")
    sys.exit(1)

File: analyze_excel.js
import XLSX from 'xlsx';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

try {
    const filePath = path.join(__dirname, 'modelo_indicadores', 'g1_prazo_decorrido.xlsx');
    console.log(`📂 Analisando arquivo: ${filePath}\n`);
    
    const wb = XLSX.readFile(filePath);
    
    console.log(`✅ Arquivo carregado com sucesso!`);
    console.log(`\n📋 Abas encontradas: ${wb.SheetNames.join(', ')}`);
    console.log(`   Total de abas: ${wb.SheetNames.length}\n`);
    
    // Verificar se tem as 3 abas necessárias
    const hasConfig = wb.SheetNames.includes('Configurações') || wb.SheetNames.includes('Configuracoes');
    const hasDados = wb.SheetNames.includes('Dados');
    const hasCores = wb.SheetNames.includes('Cores');
    
    console.log(`\n🔍 VALIDAÇÃO DO FORMATO:`);
    console.log(`   ${hasConfig ? '✅' : '❌'} Aba "Configurações" ou "Configuracoes"`);
    console.log(`   ${hasDados ? '✅' : '❌'} Aba "Dados"`);
    console.log(`   ${hasCores ? '✅' : '⚠️ '} Aba "Cores" (opcional)`);
    
    if (hasConfig && hasDados) {
        console.log(`\n✅ FORMATO CORRETO! O arquivo está no formato de 3 abas esperado.\n`);
    } else {
        console.log(`\n❌ FORMATO INCORRETO! O arquivo não está no formato esperado.\n`);
    }
    
    // Mostrar conteúdo de cada aba
    wb.SheetNames.forEach(sheetName => {
        console.log(`\n${'='.repeat(70)}`);
        console.log(`📄 ABA: "${sheetName}"`);
        console.log(`${'='.repeat(70)}`);
        
        const sheet = wb.Sheets[sheetName];
        const data = XLSX.utils.sheet_to_json(sheet, { header: 1, defval: '' });
        
        // Mostrar primeiras 15 linhas
        data.slice(0, 15).forEach((row, idx) => {
            const filteredRow = row.filter(cell => cell !== '');
            if (filteredRow.length > 0) {
                console.log(`Linha ${idx + 1}: [${filteredRow.map(c => `"${c}"`).join(', ')}]`);
            }
        });
        
        console.log(`\nTotal de linhas: ${data.length}`);
        console.log(`Total de colunas: ${data[0] ? data[0].length : 0}`);
    });
    
    console.log(`\n${'='.repeat(70)}`);
    console.log(`📊 ANÁLISE COMPLETA`);
    console.log(`${'='.repeat(70)}\n`);
    
} catch (error) {
    console.error(`❌ Erro ao analisar arquivo: ${error.message}`);
    process.exit(1);
}
